File: apps/web/types/atsCheck.ts
export type AtsRating =
  | "mukemmel"
  | "iyi"
  | "orta"
  | "iyilestirilebilir"
  | "iyilestirilmeli";

export type AtsCategoryKey = "tasarim" | "duzen" | "icerik";

export interface AtsCategoryScore {
  score: number;
  rating: AtsRating;
  feedback: string;
}

export interface AtsCheckResponse {
  overall_score: number;
  overall_rating: AtsRating;
  categories: Record<AtsCategoryKey, AtsCategoryScore>;
  summary: string;
  suggestions: string[];
}

File: apps/api/app/schemas/ats_check.py
"""Misafir ATS CV uyumluluk kontrolü yanıt şemaları."""
from typing import Literal

from pydantic import BaseModel, Field

AtsRating = Literal[
    "mukemmel",
    "iyi",
    "orta",
    "iyilestirilebilir",
    "iyilestirilmeli",
]


class AtsCategoryScore(BaseModel):
    score: int = Field(..., ge=0, le=100)
    rating: AtsRating
    feedback: str = ""


class AtsCheckResponse(BaseModel):
    overall_score: int = Field(..., ge=0, le=100)
    overall_rating: AtsRating
    categories: dict[str, AtsCategoryScore]
    summary: str = ""
    suggestions: list[str] = Field(default_factory=list)
